src/include/transaction/transaction_context.h

```diff
@@ -30,6 +31,28 @@ namespace terrier::transaction {
  */
 class TransactionContext {
  public:
+  class DebugLock : public std::shared_mutex {
+   public:
+
+    void debug_lock_shared(const common::ManagedPointer<transaction::TransactionContext> &txn) {
+      TERRIER_ASSERT(std::count(readers.begin(), readers.end(), txn) == 0, "Locking a lock I already have!");
+      readers.push_back(txn);
+      lockers.insert(txn);
+      std::shared_mutex::lock_shared();
+    }
+
+    void debug_unlock_shared(const common::ManagedPointer<transaction::TransactionContext> &txn) {
+      int UNUSED_ATTRIBUTE cnt = lockers.count(txn);
```

Comment:
dead code

Reply:
Moot point - this is a debugging class anyways